common/db_store/admin_db.py

```diff
@@ -0,0 +1,181 @@
+from enum import Enum
+import pymongo
+from abc import ABC, abstractmethod
+from common.db_store.error import DBError
+
+
+class DBAdmin(ABC):
+
+    @abstractmethod
+    def create_index(self, data_store_name, index_key, **kwargs):
+        pass
+
+    @abstractmethod
+    def list_indexes(self, data_store_name, **kwargs):
+        pass
+
+    @abstractmethod
+    def drop_index(self, data_store_name, index_name, **kwargs):
+        pass
+
+    @abstractmethod
+    def open_connection(self):
+        pass
+
+    @abstractmethod
+    def close_connection(self):
+        pass
+
+    @abstractmethod
+    def create_data_store(self, data_store_name, data_store_type, **kwargs):
+        pass
+
+    @abstractmethod
+    def delete_data_store(self, data_store_name, data_store_type, **kwargs):
+        pass
+
+    @abstractmethod
+    def get_data_stores(self):
+        pass
+
+
+class DATASTORE(Enum):
+    TIMESERIES_STORE = "timeseries_store"
```

Comment:
Can remove "_STORE" from all fields. The name "DATASTORE" itself suggests that all entries are stores.

Reply:
removed '_store' from all field values.